sdk/core/azure-core-amqp/src/main/java/com/azure/core/amqp/models/AmqpAnnotatedMessage.java

```diff
@@ -59,12 +69,15 @@ public AmqpAnnotatedMessage(AmqpAnnotatedMessage message) {
         AmqpMessageBodyType bodyType = message.getBody().getBodyType();
         switch (bodyType) {
             case DATA:
-                amqpMessageBody = AmqpMessageBody.fromData(message.getBody().getFirstData());
+                final byte[] data = message.getBody().getFirstData();
+                amqpMessageBody = AmqpMessageBody.fromData(Arrays.copyOf(data, data.length));
                 break;
             case SEQUENCE:
             case VALUE:
-                throw logger.logExceptionAsError(new UnsupportedOperationException("Body type not supported yet "
-                    + bodyType.toString()));
+                throw logger.logExceptionAsError(new UnsupportedOperationException(
+                    String.format(Locale.US, "This constructor only support body type [%s] at present. Track "
+                        + "this issue, https://github.com/Azure/azure-sdk-for-java/issues/17614 for other body type "
+                        + "support in future.", AmqpMessageBodyType.DATA.toString())));
```

Comment:
"support body type [%s] at present" -> "supports the AMQP Data body type at present"